src/bin/rdcore/rootmap.rs

```diff
@@ -218,3 +218,62 @@ fn get_luks_uuid(device: &Path) -> Result<String> {
         .trim()
         .into())
 }
+
+pub fn bind_boot(config: &BindBootConfig) -> Result<()> {
+    let boot_mount = Mount::from_existing(&config.boot_mount)?;
+    let root_mount = Mount::from_existing(&config.root_mount)?;
+    let boot_uuid = boot_mount.get_filesystem_uuid()?;
+    let root_uuid = root_mount.get_filesystem_uuid()?;
+
+    let kargs = vec![format!("boot=UUID={}", boot_uuid)];
+    visit_bls_entry_options(boot_mount.mountpoint(), |orig_options: &str| {
+        if !orig_options.starts_with("boot=") && !orig_options.contains(" boot=") {
+            bls_entry_options_delete_and_append_kargs(orig_options, &[], &kargs, &[])
+        } else {
+            Ok(None)
+        }
+    })
+    .context("appending boot kargs")?;
+
+    // put it in /run also for the first boot real root mount
+    // https://github.com/coreos/fedora-coreos-config/blob/8661649009/overlay.d/05core/usr/lib/systemd/system-generators/coreos-boot-mount-generator#L105-L108
+    let boot_uuid_run = Path::new("/run/coreos/bootfs_uuid");
+    std::fs::create_dir_all(boot_uuid_run.parent().unwrap())?;
+    std::fs::write(boot_uuid_run, &boot_uuid)?;
+
+    // bind rootfs to bootfs
+    let root_uuid_stamp = boot_mount.mountpoint().join(".root_uuid");
+    if root_uuid_stamp.exists() {
+        let bound_root_uuid = std::fs::read_to_string(root_uuid_stamp)?;
+        // Let it slide if it already matches the rootfs... that shouldn't happen unless the user
+        // is trying to force a rerun of Ignition. In that case, we'll have nicer errors and
+        // warnings elsewhere.
+        if bound_root_uuid != root_uuid {
+            bail!(
+                "boot filesystem already bound to a root filesystem (UUID: {})",
+                bound_root_uuid
+            );
+        }
+    } else {
+        std::fs::write(root_uuid_stamp, root_uuid)?;
```

Comment:
This block assumes the stamp file isn't newline-terminated, which is slightly unfriendly for debugging.